packages/compiler/src/core/helpers/progress-logger.ts

```diff
@@ -0,0 +1,125 @@
+import { Direction } from "readline";
+import { LogSink, ProcessedLog } from "../types.js";
+
+export class ConsoleSink implements LogSink {
+  log(log: ProcessedLog): void {
+    process.stdout.write(log.message);
```

Comment:
i think this approach is still going to fail in teh playground/other environment that don't have access to process.

what I was thinking is that the logger/logSink itself which is abstracted per platform is what handles the spinner and then you can just have a trackprocess/action function exposed there too